classes/poster.php

```diff
@@ -0,0 +1,118 @@
+<?php
+// This file is part of mod_datalynx for Moodle - http://moodle.org/
+//
+// It is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// It is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ *
+ * @package mod_poster
+ * @copyright based on the work by 2015 David Mudrak <david@moodle.com>{
+ *
+ * @copyright 2019 by Harald, David und Nicklas @devcamp19
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_poster;
+
+use mod_poster\event\course_module_viewed;
+use completion_info;
+
+defined('MOODLE_INTERNAL') or die();
+
+/**
+ * Class poster
+ *
+ * @package mod_datalynx
+ */
+class poster {
+
+    /**
+     * @var mixed|object|false fetched from get_record.
```

Comment:
This should be just `@var object|false` (which I prefer) or `@var mixed` but not both imho